hpke_test.go

```diff
@@ -15,118 +15,143 @@ var (
-		}
+	skI, pkI, err := suite.KEM.GenerateKeyPair(rand.Reader)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error generating initiator DH key pair: %s", kemID, kdfID, aeadID, err)
+	}
 
-		enc, ctxI, err := SetupAuthI(suite, rand.Reader, pkR, skI, info)
-		if err != nil {
-			t.Fatalf("[%d] Error in SetupAuthI: %s", id, err)
-		}
+	skR, pkR, err := suite.KEM.GenerateKeyPair(rand.Reader)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error generating responder DH key pair: %s", kemID, kdfID, aeadID, err)
+	}
 
-		ctxR, err := SetupAuthR(suite, skR, pkI, enc, info)
-		if err != nil {
-			t.Fatalf("[%d] Error in SetupBaseI: %s", id, err)
-		}
+	enc, ctxI, err := SetupAuthI(suite, rand.Reader, pkR, skI, info)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error in SetupAuthI: %s", kemID, kdfID, aeadID, err)
+	}
 
-		roundTrip(t, id, ctxI, ctxR)
+	ctxR, err := SetupAuthR(suite, skR, pkI, enc, info)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error in SetupBaseI: %s", kemID, kdfID, aeadID, err)
 	}
+
+	roundTrip(t, kemID, kdfID, aeadID, ctxI, ctxR)
 }
 
-func TestPSKAuth(t *testing.T) {
-	for id, suite := range ciphersuites {
-		if _, ok := suite.KEM.(AuthKEMScheme); !ok {
-			continue
-		}
+func roundTripPSKAuth(t *testing.T, kemID, kdfID, aeadID uint16) {
+	suite, err := GetRegisteredCipherSuite(kemID, kdfID, aeadID)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error looking up ciphersuite: %s", kemID, kdfID, aeadID, err)
+	}
 
-		skI, pkI, err := suite.KEM.GenerateKeyPair(rand.Reader)
-		if err != nil {
-			t.Fatalf("[%d] Error generating initiator DH key pair: %s", id, err)
-		}
+	if _, ok := suite.KEM.(AuthKEMScheme); !ok {
+		return
+	}
 
-		skR, pkR, err := suite.KEM.GenerateKeyPair(rand.Reader)
-		if err != nil {
-			t.Fatalf("[%d] Error generating responder DH key pair: %s", id, err)
-		}
+	skI, pkI, err := suite.KEM.GenerateKeyPair(rand.Reader)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error generating initiator DH key pair: %s", kemID, kdfID, aeadID, err)
+	}
 
-		enc, ctxI, err := SetupPSKAuthI(suite, rand.Reader, pkR, skI, psk, pskID, info)
-		if err != nil {
-			t.Fatalf("[%d] Error in SetupPSKAuthI: %s", id, err)
-		}
+	skR, pkR, err := suite.KEM.GenerateKeyPair(rand.Reader)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error generating responder DH key pair: %s", kemID, kdfID, aeadID, err)
+	}
 
-		ctxR, err := SetupPSKAuthR(suite, skR, pkI, enc, psk, pskID, info)
-		if err != nil {
-			t.Fatalf("[%d] Error in SetupBaseI: %s", id, err)
-		}
+	enc, ctxI, err := SetupPSKAuthI(suite, rand.Reader, pkR, skI, psk, pskID, info)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error in SetupPSKAuthI: %s", kemID, kdfID, aeadID, err)
+	}
+
+	ctxR, err := SetupPSKAuthR(suite, skR, pkI, enc, psk, pskID, info)
+	if err != nil {
+		t.Fatalf("[%x, %x, %x] Error in SetupBaseI: %s", kemID, kdfID, aeadID, err)
+	}
 
-		roundTrip(t, id, ctxI, ctxR)
+	roundTrip(t, kemID, kdfID, aeadID, ctxI, ctxR)
+}
+
+func TestModes(t *testing.T) {
+	for kemID, _ := range kems {
```

Comment:
Do we really want to allow for all possible combinations?

Reply:
We probably don't want to allow it by specification, but testing them here seems fine.